test/gtest/ucp/test_ucp_am.cc

```diff
@@ -402,6 +402,51 @@ class test_ucp_am_nbx : public test_ucp_am_base {
         }
     }
 
+    void test_recv_on_closed_ep(size_t size, unsigned flags = 0,
+                                bool poke_rx_progress = false,
+                                bool rx_expected = false)
+    {
+        skip_loopback();
+        test_am_send_recv(0, max_am_hdr()); // warmup wireup
+
+        m_am_received = false;
+        std::vector<char> sbuf(size, 'd');
+        ucp::data_type_desc_t sdt_desc(m_dt, &sbuf[0], size);
+
+        set_am_data_handler(receiver(), TEST_AM_NBX_ID, am_rx_check_cb, this);
+
+        ucs_status_ptr_t sreq = send_am(sdt_desc, flags);
+
+        sender().progress();
+        if (poke_rx_progress) {
+            receiver().progress();
+            if (m_am_received) {
+                request_wait(sreq);
+                UCS_TEST_SKIP_R("received all AMs before ep close");
+            }
+        }
+
+        void *close_req = receiver().disconnect_nb(0, 0,
+                                                   UCP_EP_CLOSE_MODE_FLUSH);
+        ucs_time_t deadline = ucs::get_deadline(10);
+        while (!is_request_completed(close_req) &&
+               (ucs_get_time() < deadline)) {
+            progress();
+        };
+
+        receiver().close_ep_req_free(close_req);
+
+        if (rx_expected) {
+            wait_for_flag(&m_am_received);
+        }
+        EXPECT_EQ(rx_expected, m_am_received);
+
+        // Send request may complete with error
```

Comment:
```
if (!rx_expected)  {
     // Send request may complete with error
    // (rndv should complete with EP_TIMEOUT)
    scoped_log_handler wrap_err(wrap_errors_logger);
    request_wait(sreq);
} else {
    request_wait(sreq);
}
```
since when no REPLY AM proto, we have to receive and no error should be reported, right?

Reply:
fixed